rendermodules/stack/swap_zs.py

```diff
@@ -0,0 +1,104 @@
+
+    def swap_zs_from_stacks(self, source_stack, target_stack, zvalues):
+        minz = min(zvalues)
+        maxz = max(zvalues)
+        # copy the section from target to a temp stack
+        temp_stack = "em_2d_montage_solved_swap_temp_zs_{}_ze_{}_t{}".format(int(minz), int(maxz), time.strftime("%m%d%y_%H%M%S"))
+
+        renderapi.stack.clone_stack(target_stack, temp_stack, zvalues, render=self.render)
+
+        # set target stack in loading state
+        renderapi.stack.set_stack_state(target_stack, state='LOADING', render=self.render)
+
+        # delete those sections from the target stack
+        for z in zvalues:
+            renderapi.stack.delete_section(target_stack, z, render=self.render)
+        
+        # copy sections from source stack to target stack
+        renderapi.stack.clone_stack(source_stack, target_stack, zs=zvalues, render=self.render)
+        
+        # set source stack in loading state
+        renderapi.stack.set_stack_state(source_stack, state='LOADING', render=self.render)
+        
+        # delete those sections from source stack
+        for z in zvalues:
+            renderapi.stack.delete_section(source_stack, z, render=self.render)
+        
+        # copy sections from temp stack to source stack
+        renderapi.stack.clone_stack(temp_stack, source_stack, zs=zvalues, render=self.render)
+
+        if self.args['complete_source_stack']:
+            renderapi.stack.set_stack_state(source_stack, state='COMPLETE', render=self.render)
+        
+        if self.args['complete_target_stack']:
+            renderapi.stack.set_stack_state(target_stack, state='COMPLETE', render=self.render)
+
+        # delete temp stack
+        renderapi.stack.delete_stack(temp_stack, render=self.render)
+
+        # delete source stack if specified in input
+        if self.args['delete_source_stack']:
+            renderapi.stack.delete_stack(source_stack, render=self.render)
+
+        
+
+    def run(self):
+        if len(self.args['source_stack']) != len(self.args['target_stack']):
+            raise RenderModuleException("Count of source and target stacks do not match")
+        
+        list_of_stacks = renderapi.render.get_stacks_by_owner_project(owner=self.render.DEFAULT_OWNER,
+                                                                      project=self.render.DEFAULT_PROJECT,
+                                                                      host=self.render.DEFAULT_HOST,
+                                                                      port=self.render.DEFAULT_PORT,
+                                                                      render=self.render)
+        
+        source_stacks = []
+        target_stacks = []
+        zvalues = []
+        for source_stack, target_stack, zVals in zip(self.args['source_stack'], self.args['target_stack'], self.args['zValues']):
+            if source_stack not in list_of_stacks or target_stack not in list_of_stacks:
+                continue
```

Comment:
if the user provides a stack that isn't there don't we want to raise an exception rather than just completing doing nothing?

Reply:
Yes, but no here as this is a list of source and target stacks... we do not want to stop processing the list if we did not find one pair of source and target stacks. 
The output json will include the list of successfully processed source and target stacks for bookkeeping purposes. 

We could add a print statement if we want... But that's only for logging purposes

Reply:
so if both source and target are empty should we raise an exception then?

Reply:
err.. sorry.. after the for loop if the source_stacks and target_stacks are still empty.